django-backend/fecfiler/transactions/views.py

```diff
@@ -356,6 +384,59 @@ def save_transaction(self, transaction_data, request):
         update_dependent_children(transaction_instance)
         delete_carried_forward_loans_if_needed(transaction_instance, committee_id)
         delete_carried_forward_debts_if_needed(transaction_instance, committee_id)
+
+        # Set the earlier date in order to detect when a transaction has moved forward
+        if original_date and original_date < schedule_instance.get_date():
+            transactions_after_original = Transaction.objects.get_queryset().filter(
```

Comment:
I like the idea of using django's queryset to build a common result set.  The issue i see here is that you'll need to use a different set than how you've defined `transactions_after_original` as a base for finding the next transaction by election, since the next transaction by election doesn't care who contact_1 is.

I might sugest something like:
```
transactions_after_original = (queryset..).filter(same year, after-or-created-after)
next_transaction_by_entity = transactions_after_original.filter(contact_1=original_contact_1)[:0]
if schedule e:
  next_transaction_by_election = transactions_after_original.filter(election matching stuff)[:0]
...
```

The nice thing about election aggregate is that it only needs to happen over schedule e transactions

Reply:
The set of election transactions is no longer a subset of `transactions_after_original` and I've cut it down to just schedule e transactions